gengokit/gengokit.go

```diff
@@ -160,134 +265,64 @@ func (g *generator) updateClientMethods(clientHandler io.Reader, svcFuncs []stri
-	// which should exist within the handler files
-	var serviceFunctions []string
-	for _, meth := range g.templateExec.Service.Methods {
-		serviceFunctions = append(serviceFunctions, meth.GetName())
+// serviceFunctionNames returns a slice of function names which are in the
+// definition files plus the function "NewService". Used for inserting and
+// removing functions from previously generated handler files
+func serviceFunctionsNames(meths []*deftree.ServiceMethod) []string {
+	var svcFuncs []string
+	for _, meth := range meths {
+		svcFuncs = append(svcFuncs, meth.GetName())
 	}
-	serviceFunctions = append(serviceFunctions, "NewBasicService")
-
-	serverHandlerFile := getFileByName(packageName+serverHandlerFilePath, previousFiles)
-	clientHandlerFile := getFileByName(packageName+clientHandlerFilePath, previousFiles)
-
-	for _, templateFilePath := range templateFileAssets.AssetNames() {
-		if filepath.Ext(templateFilePath) != ".gotemplate" {
-			log.WithField("Template file", templateFilePath).Debug("Skipping rendering non-buildable partial template")
-			continue
-		}
-
-		var generatedFilePath string
-		var generatedCode *bytes.Buffer
-		var err error
-
-		if templateFilePath == "NAME"+serverHandlerFilePath+"template" && serverHandlerFile != nil {
-			// If there's an existing service file, update its contents
+	svcFuncs = append(svcFuncs, "NewService")
 
-			generatedCode, err = g.updateServerMethods(serverHandlerFile, serviceFunctions)
-
-			if err != nil {
-				return nil, errors.Wrap(err, "could not modifiy service handler file")
-			}
-
-		} else if templateFilePath == "NAME"+clientHandlerFilePath+"template" && clientHandlerFile != nil {
-			// If there's an existing client_handler file, update its contents
-
-			generatedCode, err = g.updateClientMethods(clientHandlerFile, serviceFunctions)
-
-			if err != nil {
-				return nil, errors.Wrap(err, "could not modifiy client handler file")
-			}
-
-		} else {
-
-			generatedCode, err = applyTemplateFromPath(templateFilePath, g.templateExec)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not render template")
-			}
-		}
-
-		generatedFilePath = templateFilePath
-		// Change file path from .gotemplate to .go
-		generatedFilePath = strings.TrimSuffix(generatedFilePath, "template")
-
-		// Turn code buffer into string and format it
-		formattedCode := formatCode(generatedCode.Bytes())
-
-		var resp truss.SimpleFile
+	return svcFuncs
+}
 
-		// Switch "NAME" in path with packageName.
-		//i.e. for packageName = addsvc; /NAME-service/NAME-server -> /addsvc-service/addsvc-server
-		resp.Path = strings.Replace(generatedFilePath, "NAME", packageName, -1)
+// trimTemplateExecutorServiceFuncs removes functions in funcsInFile from the
+// passed templateExecutor and returns a pointer to that templateExecutor
+func trimTemplateExecutorServiceFuncs(te templateExecutor, funcsInFile map[string]bool) (*templateExecutor, error) {
```

Comment:
If the point of the function is to trim the service functions, then I think it should *just* it should trim the service functions. As in, it should accept a slice of `ServiceMethod`s, then return a different slice which is then added back on to the template executor.

If the goal though is to invisibly mutate the state of the `TemplateExecutor`, then I think `trimTemplateExecutorServiceFuncs` should probably be a method on `TemplateExecutor`.